packages/api/src/routes/internal/medical/patient-job.ts

```diff
@@ -51,6 +143,52 @@ router.post(
   })
 );
 
+const failOrCancelSchema = z.object({
+  reason: z.string(),
+});
+
+/**
+ * POST /internal/patient/job/:jobId/fail
+ *
+ * Fails the job.
+ * @param req.query.cxId - The CX ID.
+ * @param req.params.jobId - The job ID.
+ * @param req.body.reason - The reason for failing the job.
+ * @returns 200 OK
+ */
+router.post(
+  "/:jobId/fail",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const jobId = getFrom("params").orFail("jobId", req);
+    const { reason } = failOrCancelSchema.parse(req.body);
+    await failPatientJob({ jobId, cxId, reason });
+    return res.sendStatus(httpStatus.OK);
+  })
+);
+
+/**
+ * POST /internal/patient/job/:jobId/cancel
+ *
+ * Cancels the job.
+ * @param req.query.cxId - The CX ID.
+ * @param req.params.jobId - The job ID.
+ * @param req.body.reason - The reason for cancelling the job.
+ * @returns 200 OK
+ */
+router.post(
+  "/:jobId/cancel",
```

Comment:
`DELETE /:jobId` ?

Or just POST `/job/:id` with a status `cancelled`

Reply:
I mean more using HTTP methods to represent operations on entities as much as possible (RESTful).

I was thinking the client can call delete on the job, but the job service doesn't delete it, just mark it as cancelled.

But NABD.

WRT updating status, I'd rather have `POST /job/:id` and pass the status - with the command validating a state machine for the status, so we can update it to failed, cancelled, etc., instead of having diff endpoints for each updatable status. Thoughts?